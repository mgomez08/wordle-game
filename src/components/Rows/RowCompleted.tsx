import checkLetter from "../../helpers/checkLetter";
import { Box } from "../Box/Box";

import style from "./row.module.scss";
interface RowCompletedProps {
  word: string;
  solution: string;
}

export const RowCompleted = ({ word, solution }: RowCompletedProps) => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={style.row}>
      {arrayRows.map((_, index) => (
        <Box
          key={index}
          value={word[index]}
          status={checkLetter(word[index], index, solution)}
        />
      ))}
    </div>
  );
};
