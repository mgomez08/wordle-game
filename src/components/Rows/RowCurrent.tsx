import { Box } from "../Box/Box";
import styles from "./row.module.scss";

interface RowCurrentProps {
  word: string;
}

export const RowCurrent = ({ word }: RowCurrentProps) => {
  const wordArray = word.split("");
  return (
    <div className={styles.row}>
      {wordArray.map((letter, index) => (
        <Box key={index} value={letter} status="edit" />
      ))}
      {Array.from(Array(5 - wordArray.length)).map((letter, index) => (
        <Box key={index} value={""} status="edit" />
      ))}
    </div>
  );
};
