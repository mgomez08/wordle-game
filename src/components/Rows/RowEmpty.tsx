import { Box } from "../Box/Box";
import styles from "./row.module.scss";

export const RowEmpty = () => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={styles.row}>
      {arrayRows.map((_, index) => (
        <Box key={index} value="" status="empty" />
      ))}
    </div>
  );
};
