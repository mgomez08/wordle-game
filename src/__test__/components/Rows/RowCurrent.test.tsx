import { mount } from "enzyme";
import { RowCurrent } from "../../../components/Rows/RowCurrent";

describe("Tests for <RowCurrent />", () => {
  const rowcurrent = mount(<RowCurrent word="HELLO" />);

  test("Render RowCurrent component", () => {
    expect(rowcurrent.length).toEqual(1);
  });

  test("RowCurrent component should have correct number of children (boxes)", () => {
    expect(rowcurrent.find("Box").length).toEqual(5);
  });

  test("First Box children should have class (.edit)", () => {
    expect(rowcurrent.find("Box").first().find(".edit").length).toEqual(1);
  });

  test("First Box children should have 'H' in the content", () => {
    expect(rowcurrent.find("Box").first().contains("H")).toEqual(true);
  });

  test("RowCurrent component with Snapshot", () => {
    expect(rowcurrent).toMatchSnapshot();
  });
});
