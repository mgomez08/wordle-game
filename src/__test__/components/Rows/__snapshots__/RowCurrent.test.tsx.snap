// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RowCurrent /> RowCurrent component with Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="edit"
  >
    H
  </div>
  <div
    className="edit"
  >
    E
  </div>
  <div
    className="edit"
  >
    L
  </div>
  <div
    className="edit"
  >
    L
  </div>
  <div
    className="edit"
  >
    O
  </div>
</div>
`;
